### PR TITLE
Try parallel Python builds.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This parallelizes Python build into 3
=> Small & serve & dashboard & cpp tests
=> medium a-j
=> medium j-z

Result:
Python tests are done within 40 minutes. When there are errors, it can take up to an hour. It seems like Tune tests still take an hour as well. 

This won't make the whole build process faster because of Windows (which takes 1 hour and 40 minutes), but make regular waiting time for python tests much shorter.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by rkooo567** [comment]
cc @simon-mo Please give me some input

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,34 @@ matrix:
   include:
     - os: linux
       env:
-        - PYTHON=3.6
+        - PYTHON=3.6 SMALL_LARGE_TESTS=1
```
Comment (simon-mo) [inline .travis.yml:25]: SMALL_AND_LARGE_TESTS

File: .travis.yml [hunk 2]
```diff
@@ -22,7 +22,34 @@ matrix:
...
+        - RAY_CYTHON_EXAMPLES=1
+        - RAY_USE_RANDOM_PORTS=1
+      install:
+        - . ./ci/travis/ci.sh init RAY_CI_SERVE_AFFECTED,RAY_CI_TUNE_AFFECTED,RAY_CI_PYTHON_AFFECTED
+      before_script:
+        - . ./ci/travis/ci.sh build
+
+    - os: linux
+      env:
+        - PYTHON=3.6 MED_TEST_A_TO_J=1
```
Comment (simon-mo) [inline .travis.yml:37]: MEDIUM_TESTS_A_TO_J

**Review by simon-mo — APPROVED** [review]
LGTM mostly naming things

**Comment by rkooo567** [comment]
@barakmich I merged it now. Please lmk if you have any concern about this. I can still revert or create a followup PR

**Pull request merged**